TestAdd Failure

ipfscluster_test.go:347: error waiting to become a Voter: context canceled

or error waiting for Leader: context canceled

Flaky test, but happens quite often

--- a/config_test.go
+++ b/config_test.go
@@ -34,8 +34,8 @@ var testingClusterCfg = []byte(`{
     "state_sync_interval": "1m0s",
     "ipfs_sync_interval": "2m10s",
     "replication_factor": -1,
-    "monitor_ping_interval": "250ms",
-    "peer_watch_interval": "100ms",
+    "monitor_ping_interval": "350ms",
+    "peer_watch_interval": "200ms",
     "disable_repinning": false
 }`)
 
@@ -46,14 +46,14 @@ var testingRaftCfg = []byte(`{
     "commit_retry_delay": "50ms",
     "backups_rotate": 2,
     "network_timeout": "5s",
-    "heartbeat_timeout": "100ms",
-    "election_timeout": "100ms",
-    "commit_timeout": "50ms",
+    "heartbeat_timeout": "200ms",
+    "election_timeout": "200ms",
+    "commit_timeout": "150ms",
     "max_append_entries": 256,
     "trailing_logs": 10240,
     "snapshot_interval": "2m0s",
     "snapshot_threshold": 8192,
-    "leader_lease_timeout": "80ms"
+    "leader_lease_timeout": "200ms"
 }`)
 
 var testingCrdtCfg = []byte(`{
@@ -110,12 +110,12 @@ var testingTrackerCfg = []byte(`
 `)
 
 var testingMonCfg = []byte(`{
-    "check_interval": "400ms",
+    "check_interval": "550ms",
     "failure_threshold": 6
 }`)
 
 var testingDiskInfCfg = []byte(`{
-    "metric_ttl": "250ms",
+    "metric_ttl": "350ms",
     "metric_type": "freespace"
 }`)
 

--- a/consensus/raft/raft.go
+++ b/consensus/raft/raft.go
@@ -40,7 +40,7 @@ var RaftLogCacheSize = 512
 
 // How long we wait for updates during shutdown before snapshotting
 var waitForUpdatesShutdownTimeout = 5 * time.Second
-var waitForUpdatesInterval = 100 * time.Millisecond
+var waitForUpdatesInterval = 400 * time.Millisecond
 
 // How many times to retry snapshotting when shutting down
 var maxShutdownSnapshotRetries = 5
@@ -283,6 +283,7 @@ func (rw *raftWrapper) WaitForVoter(ctx context.Context) error {
 		case <-ctx.Done():
 			return ctx.Err()
 		default:
+			logger.Debugf("%s: get configuration", pid)
 			configFuture := rw.raft.GetConfiguration()
 			if err := configFuture.Error(); err != nil {
 				return err
@@ -291,6 +292,7 @@ func (rw *raftWrapper) WaitForVoter(ctx context.Context) error {
 			if isVoter(pid, configFuture.Configuration()) {
 				return nil
 			}
+			logger.Debugf("%s: not voter yet", pid)
 
 			time.Sleep(waitForUpdatesInterval)
 		}

